lib/services/file_magic_service.dart

```diff
@@ -0,0 +1,145 @@
+import 'dart:convert';
+
+import 'package:dio/dio.dart';
+import 'package:flutter_sodium/flutter_sodium.dart';
+import 'package:logging/logging.dart';
+import 'package:photos/core/event_bus.dart';
+import 'package:photos/core/network.dart';
+import 'package:photos/db/files_db.dart';
+import 'package:photos/events/files_updated_event.dart';
+import 'package:photos/models/file.dart';
+import 'package:photos/core/configuration.dart';
+import 'package:photos/models/file_magic_metadata.dart';
+import 'package:photos/services/remote_sync_service.dart';
+
+import '../utils/crypto_util.dart';
+import '../utils/file_download_util.dart';
+
+class FileMagicService {
+  final _logger = Logger("FileMagicService");
+  Dio _dio;
+  FilesDB _filesDB;
+
+  FileMagicService._privateConstructor() {
+    _filesDB = FilesDB.instance;
+    _dio = Network.instance.getDio();
+  }
+
+  static final FileMagicService instance =
+      FileMagicService._privateConstructor();
+
+  Future<void> changeVisibility(List<File> files, int visibility) async {
+    Map<String, dynamic> update = {};
+    update[kMagicKeyVisibility] = visibility;
+    return _updateMagicData(files, update);
+  }
+
+  Future<void> _updateMagicData(
+      List<File> files, Map<String, dynamic> newMetadataUpdate) async {
+    final params = <String, dynamic>{};
+    params['metadataList'] = [];
+    int ownerID = Configuration.instance.getUserID();
+    try {
+      for (final file in files) {
+        if (file.uploadedFileID == null) {
+          throw AssertionError(
+              "operation is only supported on backed up files");
+        } else if (file.ownerID != ownerID) {
+          throw AssertionError("can not modify memories not owned by you");
```

Comment:
`cannot modify memories not owned by you` (remove space, assuming this could be surfaced on the UI)